fix: mobile input not working

0NotApplicable0 opened the pull request:
Fixes: https://github.com/bony2023/react-terminal/issues/234

"Fixes" however there are some downsides to this. Because browsers require an event to be fired to control anything we can no longer set "consoleFocused" to true to allow a virtual keyboard to show. I think this is just a limitation of devices?

Second, I overwrite the parent wrapper's onClick event which isn't being used currently but if it is modified in the future we will have to change this logic.

I also downgraded react-device-detect to 2.1.2 from 2.2.2 as from testing isMobile was always wrong on an actual mobile device even though it was correct on a browsers mobile simulation.

0NotApplicable0 commented:
I experimented with forwarding refs and this solution worked but added a whole not of extra code. Passing the parent ref was the simplest solution I could think of. If we ever add in onClick for the parent div then we can move the forward ref approach if we want.

0NotApplicable0 commented:
@bony2023 sure thing! I can go ahead and see if I can write a test case.

0NotApplicable0 commented:
@bony2023 I've never actually tried testing for mobile on a react app. Any idea on how to do this? More specifically, mock a mobile device? I know we can change the view port size but I don't think this is what we are looking for.

It would be pretty easy to create a test to click on the wrapperRef and check if the input is in focus but this doesn't really test if it will work on mobile.

Any ideas?

0NotApplicable0 commented:
@bony2023 I added some test cases for checking if we can focus on the input when on mobile by clicking in the editor. 
The only thing I couldn't find how to do was freeze the package version of react-device-detect on a package.json level. Maybe you have some insight? I saw ways to modify the dependabot config to ignore things but that is it.

bony2023 commented on src/components/Terminal.tsx:
```diff
@@ -46,7 +46,9 @@ export default function Terminal(props: any) {
     showControlBar={showControlBar}
-    defaultHandler={defaultHandler}/>
+    defaultHandler={defaultHandler}
+    editorStyle={props.editorStyle}
```
Can you document what this prop is used for? If it's meant to supply custom css to the editor, can it be done through custom themes?

0NotApplicable0 replied:
https://github.com/bony2023/react-terminal/pull/255#issuecomment-1371515722

bony2023 commented:
@0NotApplicable0 thanks for this. Without giving `~` or `^`, the package.json should use the exact version which is what you've done, so I think it should be fine. Dependabot should ignore newer versions. In case, it creates a bump PR for this package, I'll see if we can disable dependabot for this package separately.

bony2023 reviewed (changes requested):
The changes looks good to me. Just want to understand about the new prop and it's usage.

0NotApplicable0 commented:
@bony2023 good thing you caught that. I added that in for testing in the very beginning and its not actually being used. I went ahead and removed it. It was meant supply custom css to the editor just for my testing. Nothing related to this PR.
